### PR TITLE
Added support for template icons on macOS

### PR DESCRIPTION
**Pull request by oxtoacart**
For #45

### CODE REVIEW

File: systray_darwin.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import (
+	"unsafe"
+)
+
+// SetTemplateIcon sets the systray icon as a template icon (on Mac), falling back
+// to a regular icon on other platforms.
+// templateIconBytes and iconBytes should be the content of .ico for windows and
+// .ico/.jpg/.png for other platforms.
+func SetTemplateIcon(templateIconBytes []byte, iconBytes []byte) {
```
Comment (oxtoacart) [inline systray_darwin.go]: As described in the comment, `SetTemplateIcon` takes both a template icon and a regular icon, using the template icon on macOS and the regular icon on all other platforms.

File: systray_darwin.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"unsafe"
+)
+
+// SetTemplateIcon sets the systray icon as a template icon (on Mac), falling back
+// to a regular icon on other platforms.
+// templateIconBytes and regularIconBytes should be the content of .ico for windows and
+// .ico/.jpg/.png for other platforms.
+func SetTemplateIcon(templateIconBytes []byte, regularIconBytes []byte) {
```
Comment (oxtoacart) [inline systray_darwin.go]: As the comment states, `SetTemplateIcon` accepts both a template icon and a regular icon, using the template icon on macOS and the regular icon on other platforms.

File: systray_darwin.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	cstr := (*C.char)(unsafe.Pointer(&templateIconBytes[0]))
+	C.setIcon(cstr, (C.int)(len(templateIconBytes)), true)
+}
+
+// SetTemplateIcon sets the icon of a menu item as a template icon (on macOS). On Windows, it
+// falls back to the regular icon bytes and on Linux it does nothing.
+// templateIconBytes and regularIconBytes should be the content of .ico for windows and
+// .ico/.jpg/.png for other platforms.
+func (item *MenuItem) SetTemplateIcon(templateIconBytes []byte, regularIconBytes []byte) {
```
Comment (oxtoacart) [inline systray_darwin.go]: You can set template icons on menu items as well.

File: systray_linux.c [hunk 4]
```diff
@@ -172,7 +172,7 @@ gboolean do_quit(gpointer data) {
 	return FALSE;
 }
 
-void setIcon(const char* iconBytes, int length) {
+void setIcon(const char* iconBytes, int length, bool template) {
```
Comment (oxtoacart) [inline systray_linux.c]: new parameter is ignored on linux

File: systray_linux.c [hunk 5]
```diff
@@ -186,7 +186,7 @@ void setTooltip(char* ctooltip) {
 	free(ctooltip);
 }
 
-void setMenuItemIcon(const char* iconBytes, int length, int menuId) {
+void setMenuItemIcon(const char* iconBytes, int length, int menuId, bool template) {
```
Comment (oxtoacart) [inline systray_linux.c]: new parameter is ignored on linux

File: systray_linux.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+#include "systray.h"
+*/
+import "C"
+
+// SetTemplateIcon sets the systray icon as a template icon (on macOS), falling back
+// to a regular icon on other platforms.
+// templateIconBytes and iconBytes should be the content of .ico for windows and
+// .ico/.jpg/.png for other platforms.
+func SetTemplateIcon(templateIconBytes []byte, regularIconBytes []byte) {
+	SetIcon(regularIconBytes)
```
Comment (oxtoacart) [inline systray_linux.go]: On Linux, `SetTemplateIcon` just uses the regular icon.

File: systray_linux.go [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+// SetTemplateIcon sets the icon of a menu item as a template icon (on macOS). On Windows, it
+// falls back to the regular icon bytes and on Linux it does nothing.
+// templateIconBytes and regularIconBytes should be the content of .ico for windows and
+// .ico/.jpg/.png for other platforms.
+func (item *MenuItem) SetTemplateIcon(templateIconBytes []byte, regularIconBytes []byte) {
```
Comment (oxtoacart) [inline systray_linux.go]: On Linux, `SetTemplateIcon` does nothing.

File: systray_linux.go [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+// to a regular icon on other platforms.
+// templateIconBytes and iconBytes should be the content of .ico for windows and
+// .ico/.jpg/.png for other platforms.
+func SetTemplateIcon(templateIconBytes []byte, regularIconBytes []byte) {
+	SetIcon(regularIconBytes)
+}
+
+// SetIcon sets the icon of a menu item. Only works on macOS and Windows.
+// iconBytes should be the content of .ico/.jpg/.png
+func (item *MenuItem) SetIcon(iconBytes []byte) {
```
Comment (oxtoacart) [inline systray_linux.go]: On Linux, `SetIcon` does nothing. This is not new, this just makes it clearer.

File: systray_windows.go [hunk 9]
```diff
@@ -104,6 +104,14 @@ func SetIcon(iconBytes []byte) {
 	}
 }
 
+// SetTemplateIcon sets the systray icon as a template icon (on macOS), falling back
+// to a regular icon on other platforms.
+// templateIconBytes and iconBytes should be the content of .ico for windows and
+// .ico/.jpg/.png for other platforms.
+func SetTemplateIcon(templateIconBytes []byte, regularIconBytes []byte) {
+	SetIcon(regularIconBytes)
```
Comment (oxtoacart) [inline systray_windows.go]: On Windows, `SetTemplateIcon` just uses the regular icon.

File: example/main.go [hunk 10]
```diff
@@ -21,7 +21,7 @@ func main() {
 }
 
 func onReady() {
-	systray.SetIcon(icon.Data)
+	systray.SetTemplateIcon(icon.Data, icon.Data)
```
Comment (joesis) [inline example/main.go]: Maybe leave `SetIcon` as is in the example? I expect most users would better just use `SetIcon`.

File: systray_darwin.go [hunk 11]
```diff
@@ -0,0 +1,39 @@
+package systray
+
+/*
+#cgo linux CFLAGS: -DWEBVIEW_GTK=1
+#cgo linux pkg-config: gtk+-3.0 webkit2gtk-4.0 appindicator3-0.1
```
Comment (joesis) [inline systray_darwin.go]: linux flags are not useful here.

File: systray_linux.go [hunk 12]
```diff
@@ -0,0 +1,31 @@
+package systray
+
+/*
+#cgo linux CFLAGS: -DWEBVIEW_GTK=1
+#cgo linux pkg-config: gtk+-3.0 webkit2gtk-4.0 appindicator3-0.1
+#cgo darwin CFLAGS: -DDARWIN -x objective-c -fobjc-arc
+#cgo darwin LDFLAGS: -framework Cocoa -framework WebKit
```
Comment (joesis) [inline systray_linux.go]: darwin flags are not useful here

File: systray_darwin.go (hunk 11 above)
Comment (oxtoacart) [reply systray_darwin.go]: Thanks, will fix.

File: systray_linux.go (hunk 12 above)
Comment (oxtoacart) [reply systray_linux.go]: Thanks, will fix.

**Comment by oxtoacart** [comment]
Thanks for the review @joesis!

**Pull request merged**